README.txt
=====================================================================================================
TYPO3: Inline Relational Record Editing Tutorial
(c) 2007-2010 Oliver Hader <oliver@typo3.org> - All rights reserved
=====================================================================================================
This script is part of the TYPO3 project. The TYPO3 project is free software; you can redistribute
it and/or modify it under the terms of the GNU General Public License as published by the Free So
Software Foundation; either version 2 of the License, or (at your option) any later version.
The GNU General Public License can be found at http://www.gnu.org/copyleft/gpl.html.
=====================================================================================================


-----------------------------
 1. Installation
---------------------------
Download the extension "irre_tutorial" from the TYPO3 Extension Repository at http://www.typo3.org/
Install the extension using the Extension Manager (EM) in the TYPO3 Backend. You need administrative
rights (admin user) at your TYPO3 installation to be able to do this.

After this extension was installed, a new backend module "Web>IRRE Tutorial" will appear. By clicking
on the module link in the left menu frame, you can import a set of sample data.

To uninstall the sample data you have to select "Uninstall sample data" from the upper right corner
in the "IRRE Tutorial" module. The script looks for a page alias "irre_tutorial_data" and removes, if
successfully found, this branch from your TYPO3 installation.



-----------------------------
 2. Usage
---------------------------
After you imported the IRRE Tutorial sample data, you are can play around, change tca.php files etc.
You can just do, what ever you'd like to. The ext_tables.php and tca.php have a additional string in
their filenames. This should help you to find the case you're looking fore faster. See the following
description what these strings (e.g. like "tx_irretutorial_<string>_hotel or tca.<string>.php) mean:

	* 1ncsv: 1:n relations using comma separated values as list
	* 1nff: 1:n relations using foreign_field as pointer on child table
	* mnasym: m:n bidirectional asymmetric relations using intermediate table
	* mnsym: m:n bidirectional symmetric relations using intermediate table
	* mnattr: m:n bidirectional asymmetric attributed relations using intermediate table
	* mnmmasym: m:n bidirectional asymmetric relations using the default MM feature of TYPO3



-----------------------------
 3. To-Do
---------------------------
...



-----------------------------
 4. Links
---------------------------
Extension:	http://typo3.org/extensions/repository/view/irre_tutorial/
Thesis:		http://typo3.org/development/articles/inline-relational-record-editing/
Core API:	http://typo3.org/documentation/document-library/core-documentation/doc_core_api/current/
Wiki:		http://wiki.typo3.org/index.php/IRRE
XING:		https://www.xing.com/profile/Oliver_Hader


=====================================================================================================

ChangeLog
2013-08-17  Oliver Hader  <oliver@typo3.org>

	* [TASK] Modify abstract setUp() to be compatible with TYPO3 6.1

2013-08-15  Oliver Hader  <oliver@typo3.org>

	* [TASK] Use backend module dispatcher
	* [TASK] Use include_once() instead of require_once() for 6.2 compatibility

2013-03-03  Oliver Hader  <oliver@typo3.org>

	* [TASK] #29278: Extend test-case for 1n:FF copy localized new structure on workspaces

2013-02-05  Oliver Hader  <oliver@typo3.org>

	* [TASK] #29278: Extend test-case for 1n:FF copy actions on workspaces
	* [BUGFIX] Fix unit-test assertions (array_*_key vs. array_*_assoc)
	* [TASK] #29278: Extend test-case for copying 1n:CSV & 1n:FF to different page on workspaces

2013-02-04  Oliver Hader  <oliver@typo3.org>

	* [TASK] #45132 & #45133: Extend test-case concerning placeholder translation values in workspaces
	* [TASK] #29278 & #45132 & #45133: Extend test-cases concerning sys_refindex
	* [TASK] Disable shadowColumnsForNewPlaceholders in favour of #29278 which solves it implicitly in the Core
	* [TASK] #45151: Extend test-case for 1n:CSV copy actions on workspaces

2013-01-31  Oliver Hader  <oliver@typo3.org>

	* [TASK] Trigger localization of children on parent localization
	* [TASK] #29278: Extend test-case concerning copying and localization of new workspace records
	* [TASK] Remove checks for TYPO3 versions prior to 4.5.0
	* [TASK] #29278: Extend test-case concerning placeholder values in workspaces

2012-11-05  Oliver Hader  <oliver@typo3.org>

	* [TASK] Added FlexForm for IRRE to tt_content records
	* [TASK] Utilize foreign_match_fields to distinguish references

2012-10-12  Oliver Hader  <oliver@typo3.org>

	* [TASK] Added localization in pages_language_overlay of 1nff record

2012-03-12  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Regular non-workspace tests might be executed on a workspace

2012-03-10  Oliver Hader  <oliver@typo3.org>

	* Cleanup: Changed variables, methods, CGL and install repsonse
	* Release of version 0.4.0

2012-03-06  Oliver Hader  <oliver@typo3.org>

	* Tests: Clean-up test cases
	* Tests: Added test cases for publishing records deleted in workspace

2011-08-12  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Extensions with hooks in t3lib_TCEmain are not considered to be loaded
	* Fixed bug: Behaviour in TYPO3 4.6 concerning caches has changed and needs special handling

2011-06-20  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Typing error in install process
	* Tests: Modified test cases to clear TSconfig settings options.workspaces

2011-06-14  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Calls to deprecated and removed functions

2011-04-05  Oliver Hader  <oliver@typo3.org>

	* Tests: Separated m:n MM localization tests for localizationModes 'keep' and 'select'

2011-03-30  Oliver Hader  <oliver@typo3.org>

	* Tests: Refactored abstract test clases
	* Fixed bug: TCA of m:n MM tables does not use MM relations with uid
	* Tests: Extended m:n MM localization tests

2011-03-23  Oliver Hader  <oliver@typo3.org>

	* Tests: Added m:n MM localization tests
	* Tests: Extended m:n MM localization tests

2011-03-22  Oliver Hader  <oliver@typo3.org>

	* Tests: Refactoring of structure

2011-03-18  Oliver Hader  <oliver@typo3.org>

	* Tests: Added 1:n CSV tests for removing of child records in workspaces
	* Cleanup: Fixed whitespaces in fixtures
	* Tests: Extended 1:n CSV and 1:n FF tests for removing child records in workspaces and flushing dependent elements in the workspace module

2011-03-17  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Workspaces extension must be loaded if using TYPO3 4.5.0+
	* Tests: Added 1:n FF tests for removing of child records in workspaces

2011-02-09  Oliver Hader  <oliver@typo3.org>

	* Tests: Renamed abstract test class to not be considered as testsuite by PHPUnit
	* Tests: Added skip calls to unsupported test cases that rely on TYPO3 4.5.0

2011-01-21  Oliver Hader  <oliver@typo3.org>

	* Added tabs for forms view for all parent and child tables

2011-01-01  Oliver Hader  <oliver@typo3.org>

	* Tests: Fixed 1:n FF test case since checking on a placeholder for an existing record does not make sense

2010-12-31  Oliver Hader  <oliver@typo3.org>

	* Tests: Extended 1:n FF tests concerning sorting order of placeholder records

2010-12-26  Oliver Hader  <oliver@typo3.org>

	* Tests: Extended 1:n FF tests concerning sorting order of placeholder records

2010-12-23  Oliver Hader  <oliver@typo3.org>

	* Tests: Extended 1:n FF tests concerning child records on pages

2010-11-17  Oliver Hader  <oliver@typo3.org>

	* Release of version 0.3.2

2010-11-13  Oliver Hader  <oliver@typo3.org>

	* Tests: Fixed tests to be working with TYPO3 4.5 and also versions before

2010-11-13  Oliver Hader  <oliver@typo3.org>

	* Tests: Fixed data set of table tx_irretutorial_1nff_price
	* Tests: Extended unit tests for 1:n CSV and 1:n FF
	* Updated sample data to import at ./irre_tutorial/res/T3D__IRRE.t3d
	* Release of version 0.3.1

2010-11-05  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Inline field for pages table is not available for all doktypes

2010-11-04  Oliver Hader  <oliver@typo3.org>

	* Added parenttable field for prices in 1:n FF context

2010-10-29  Oliver Hader  <oliver@typo3.org>

	* Tests: Fixed labels of error messages to be checked

2010-10-28  Oliver Hader  <oliver@typo3.org>

	* Changed blob fields of 1:n CSV relations to text
	* Updated README.txt contents
	* Tests: Extended unit tests for 1:n FF
	* Tests: Extended unit tests for 1:n CSV

2010-10-27  Oliver Hader  <oliver@typo3.org>

	* Added TCA definition to copy foreign_field and foreign_table_field values to workspace placeholder records

2010-10-20  Oliver Hader  <oliver@typo3.org>

	* Added use case to use 1:n relationships as part of a pages record
	* Updated sample data to import at ./irre_tutorial/res/T3D__IRRE.t3d

2010-10-10  Oliver Hader  <oliver@typo3.org>

	* Added basic skeleton for unit tests
	* Tests: Extended unit tests for 1:n FF


ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "irre_tutorial".
#
# Auto generated 10-03-2012 21:03
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Tutorial for Inline Relational Record Editing IRRE',
	'description' => 'Shows how Inline Relational Record Editing could be used to build a structure of hotel, offers and prices differently. Use the "IRRE Tutorial" module to get a quickstart by installing some IRRE sample data.',
	'category' => 'example',
	'shy' => 0,
	'version' => '0.4.0',
	'dependencies' => 'workspaces,version',
	'conflicts' => '',
	'priority' => '',
	'loadOrder' => '',
	'module' => 'mod1',
	'state' => 'beta',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearcacheonload' => 0,
	'lockType' => '',
	'author' => 'Oliver Hader',
	'author_email' => 'oliver@typo3.org',
	'author_company' => '',
	'CGLcompliance' => '',
	'CGLcompliance_note' => '',
	'constraints' => array(
		'depends' => array(
			'typo3' => '4.5.0-0.0.0',
			'workspaces' => '0.0.0-',
			'version' => '0.0.0-',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:49:{s:9:"ChangeLog";s:4:"beb5";s:10:"README.txt";s:4:"e002";s:16:"ext_autoload.php";s:4:"b016";s:12:"ext_icon.gif";s:4:"b4e6";s:20:"ext_tables.1ncsv.php";s:4:"7720";s:19:"ext_tables.1nff.php";s:4:"095d";s:22:"ext_tables.general.php";s:4:"7785";s:21:"ext_tables.mnasym.php";s:4:"ba43";s:21:"ext_tables.mnattr.php";s:4:"f80c";s:23:"ext_tables.mnmmasym.php";s:4:"bf72";s:20:"ext_tables.mnsym.php";s:4:"21c5";s:14:"ext_tables.php";s:4:"ae00";s:14:"ext_tables.sql";s:4:"5e78";s:30:"icon_tx_irretutorial_hotel.gif";s:4:"4ad7";s:40:"icon_tx_irretutorial_hotel_offer_rel.gif";s:4:"d3a6";s:34:"icon_tx_irretutorial_hotel_rel.gif";s:4:"9e5c";s:30:"icon_tx_irretutorial_offer.gif";s:4:"1e24";s:40:"icon_tx_irretutorial_offer_price_rel.gif";s:4:"4025";s:30:"icon_tx_irretutorial_price.gif";s:4:"dc05";s:16:"locallang_db.xml";s:4:"3e61";s:13:"tca.1ncsv.php";s:4:"136a";s:12:"tca.1nff.php";s:4:"5887";s:14:"tca.mnasym.php";s:4:"19d6";s:14:"tca.mnattr.php";s:4:"fcdf";s:16:"tca.mnmmasym.php";s:4:"11e7";s:13:"tca.mnsym.php";s:4:"8454";s:14:"doc/manual.sxw";s:4:"b30a";s:14:"mod1/clear.gif";s:4:"cc11";s:13:"mod1/conf.php";s:4:"3f25";s:14:"mod1/index.php";s:4:"5703";s:18:"mod1/locallang.xml";s:4:"29c4";s:22:"mod1/locallang_mod.xml";s:4:"3d98";s:19:"mod1/moduleicon.gif";s:4:"b4e6";s:17:"res/T3D__IRRE.t3d";s:4:"030b";s:51:"tests/class.tx_irretutorial_1ncsvWorkspacesTest.php";s:4:"24ae";s:50:"tests/class.tx_irretutorial_1nffWorkspacesTest.php";s:4:"f45a";s:40:"tests/class.tx_irretutorial_Abstract.php";s:4:"bf1d";s:52:"tests/class.tx_irretutorial_AbstractLocalization.php";s:4:"52b5";s:50:"tests/class.tx_irretutorial_AbstractWorkspaces.php";s:4:"9473";s:60:"tests/class.tx_irretutorial_mnmmaysmLocalizationKeepTest.php";s:4:"6076";s:62:"tests/class.tx_irretutorial_mnmmaysmLocalizationSelectTest.php";s:4:"0fe6";s:29:"tests/fixtures/data_1ncsv.xml";s:4:"4e87";s:28:"tests/fixtures/data_1nff.xml";s:4:"4b5e";s:30:"tests/fixtures/data_mnasym.xml";s:4:"4f9a";s:32:"tests/fixtures/data_mnmmasym.xml";s:4:"fade";s:29:"tests/fixtures/data_mnsym.xml";s:4:"3f09";s:29:"tests/fixtures/data_pages.xml";s:4:"87ad";s:36:"tests/fixtures/data_sys_language.xml";s:4:"fb16";s:37:"tests/fixtures/data_sys_workspace.xml";s:4:"5185";}',
	'suggests' => array(
	),
);

?>